src/input/useInput.ts

```diff
@@ -67,6 +67,9 @@ export default function useInput(props: TdInputProps, expose: (exposed: Record<s
       const stringInfo = getCharacterLength(val, props.maxcharacter);
       val = typeof stringInfo === 'object' && stringInfo.characters;
     }
+    if (props.type === 'number' && typeof props.maxlength === 'number' && props.maxlength >= 0) {
+      val = val.substring(0, props.maxlength);
+    }
```

Comment:
```suggestion
    if (props.type === 'number' && props?.maxlength > 0) {
      val = val.substring(0, props.maxlength);
    }
```
`>= 0`是不对的，`值小于等于 0 的时候，则表示不限制输入长度。`
没必要判断maxlength是不是number，毕竟类型约束
![image](https://user-images.githubusercontent.com/36812439/176100844-de24ff36-9b57-49d3-87b4-b27b4cb184a1.png)

Reply:
类型校验是考虑到最终在使用的时候还是在js环境运行，使用方如果不使用ts的类型校验的话，vue只会提供一个warning级别的console并没有对类型进行强制。maxlength等于0这个是看目前input等于0的时候其实也是限制了无法输入的。才将等于0也包括进去了。感谢大佬的建议。这两个问题我们是否可以再讨论一下?

Reply:
> 类型校验是考虑到最终在使用的时候还是在js环境运行，使用方如果不使用ts的类型校验的话，vue只会提供一个warning级别的console并没有对类型进行强制。maxlength等于0这个是看目前input等于0的时候其实也是限制了无法输入的。才将等于0也包括进去了。感谢大佬的建议。这两个问题我们是否可以再讨论一下?

嗯...确实我没考虑到js,类型校验可以有，但是等于号还是不必，毕竟写`值小于等于 0 的时候，则表示不限制输入长度`如果测试下来写0也限制了那估计是BUG

Reply:
```suggestion
    if (props.type === 'number' && typeof props.maxlength === 'number' && props?.maxlength > 0) {
      val = val.substring(0, props.maxlength);
    }
```

Reply:
嗯嗯  之前写的时候就是还纠结了一下要不要包含0进去，测试过后发现maxlength使用0的时候确实限制了就给包含进去了。我给改改把等号去掉再来提交